### PR TITLE
ED25519 with certificates and TLS

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: certs/ed25519/ca-ed25519-key.pem [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (toddouska) [inline certs/ed25519/ca-ed25519-key.pem]: I don't see the new files in this PR being added to include.am(s).  Did I miss those?

Why is the PEM header "RSA KEY" while the file name is ed25519?

**Comment by SparkiDev** [comment]
The PEM header should be "BEGIN PRIVATE KEY" (https://tools.ietf.org/html/draft-ietf-curdle-pkix-04) but PemToDer converts the PKCS #8 key to a private only key. The signature scheme needs to public key in order to work. So, for now, I've used a different header.

File: certs/ed25519/ca-ed25519-key.pem (hunk 1 above)
Comment (SparkiDev) [inline certs/ed25519/ca-ed25519-key.pem]: Fixed

File: wolfcrypt/src/asn.c [hunk 2]
```diff
@@ -2140,6 +2162,41 @@ int wc_CheckPrivateKey(byte* key, word32 keySz, DecodedCert* der)
...
 
+    #ifdef HAVE_ED25519
+    {
+        int ret = 0;
+        word32  keyIdx = 0;
+        ed25519_key key_pair;
+
+        if (der->keyOID == ED25519k) {
+            if ((ret = wc_ed25519_init(&key_pair)) == 0 &&
+                wc_Ed25519PrivateKeyDecode(key, &keyIdx, &key_pair, keySz)
```
Comment (toddouska) [inline wolfcrypt/src/asn.c]: If the key decode here fails it looks like we might miss that problem.

Comment (SparkiDev) [inline wolfcrypt/src/asn.c]: reworked function

**Pull request merged**